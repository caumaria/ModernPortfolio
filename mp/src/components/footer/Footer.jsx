import { Link } from "react-scroll";
import styled from "styled-components";
import FooterSection from "./FooterSection";
import ArrowUp from '/arrowUp.svg';
import React from "react";

const FooterContainer = styled.div`
  background-color: var(--dark-green);
  background-image: url("https://www.transparenttextures.com/patterns/black-orchid.png");
  min-height: 700px;
  height: 100%;
  font-size: 1.5rem;
  position: relative;
  text-align: center;

  section {
    width: 100%;
  }

  h5 {
    color: var(--pink);
    font-size: 3rem;
    margin: 0;
    padding: 3rem 0;
  }

  pre {
    margin-top: 8rem;
    opacity: .7;
    font-size: small;
  }
`;

const TopScrollLink = styled.div`
  position: absolute;
  bottom: 6%;
  right: 3%;
  img {
    width: 50px;
    cursor: pointer;
    transition: transform 0.5s cubic-bezier(0.34, 1.56, 0.64, 1);
  }

  &:hover img {
    transform: scale(1.2);
  }

  @media (max-width: 768px) {
    padding-right: 1rem;
  }
`;

export default function Footer() {
  return (
    <FooterContainer id="contact">
      <h5>What&rsquo;s next?</h5>
      <section>
        <FooterSection />
      </section>

      <TopScrollLink>
        <Link to="home" smooth={true} duration={1200}>
          <img src={ArrowUp} alt="arrow scroll to top" />
        </Link>
      </TopScrollLink>

      <pre>Â© Cau Ugolini 2023, Dezembro</pre>
    </FooterContainer>
  );
}
