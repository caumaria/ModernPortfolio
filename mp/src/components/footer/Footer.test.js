import "@testing-library/jest-dom";
import renderer from 'react-test-renderer';
import Footer from "./Footer";
import React from "react";

describe("should render work component correctly", () => {
    it('performs snapshot testing', () => {
      expect.hasAssertions();
      const tree = renderer.create(<Footer />).toJSON();
      expect(tree).toMatchSnapshot();
    });
});
  
