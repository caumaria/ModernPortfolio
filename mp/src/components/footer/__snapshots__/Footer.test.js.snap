// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render work component correctly performs snapshot testing 1`] = `
<div
  className="sc-eCstZk dcgvxZ"
  id="contact"
>
  <h5>
    What’s next?
  </h5>
  <section>
    <div
      className="sc-hKgKIp PnJdO"
    >
      <section>
        <div
          className="sc-dlfmHC dKMaks"
        >
          <h6>
            Start by saying 
            <span>
              hi
            </span>
            !
          </h6>
          <div>
            <p>
              I love to talk about games and new technologies, feel free to contact me anytime.
            </p>
            <p>
              Feedbacks are appreciated ♥
            </p>
            <h4>
              Get in
              <a
                data-content="touch"
                href="mailto: cau.ugolini@gmail.com"
              >
                touch
              </a>
              .
            </h4>
          </div>
        </div>
        <div
          className="sc-bdfCDU jMdLsz"
        >
          <div>
            <a
              href="https://github.com/caumaria"
              target="blank"
            >
              <img
                alt="Github de Cau"
                src=""
              />
              <p>
                Github
              </p>
            </a>
            <a
              href="https://www.linkedin.com/in/cau-ugolini-923a4a185/"
              target="blank"
            >
              <img
                alt="Linked In de Cau"
                src=""
              />
              <p>
                Linked In
              </p>
            </a>
          </div>
          <a
            className="sc-gsTDqH RCTip"
            href="https://www.buymeacoffee.com/cau_"
            target="blank"
          >
            <div>
              <img
                alt="coffee cup"
                src="/coffee.jpg"
              />
            </div>
          </a>
          <h6>
            01 coffee a day keeps the bugs away...
          </h6>
        </div>
      </section>
    </div>
  </section>
  <div
    className="sc-jSguLX kTsCsb"
  >
    <a
      onClick={[Function]}
      style={{}}
    >
      <img
        alt="arrow scroll to top"
        src=""
      />
    </a>
  </div>
  <pre>
    © Cau Ugolini 2023, Dezembro
  </pre>
</div>
`;
